### ISSUE TITLE
Explore: Customized Cross Validation function in dask ml

### ISSUE THREAD

**Issue by rogerkuou** [VegeWaterDynamics/motrainer#74]
In the [example notebook](https://github.com/VegeWaterDynamics/motrainer/blob/generalization/example/dssat_s1/dask_ml.ipynb), we use the standard CV functions from sklearn. Sometimes the CV function need to be customized. One example is the [Jackknife CV](https://github.com/VegeWaterDynamics/motrainer/blob/generalization/motrainer/jackknife.py ) function. How can we specify the customized CV in dask-ml?

**Comment by fnattino** [comment]
The notebook linked above uses [GridSearchCV](https://ml.dask.org/modules/generated/dask_ml.model_selection.GridSearchCV.html#dask-ml-model-selection-gridsearchcv) from `dask-ml` to perform the search of optimal parameters. This search method is suitable for use cases for which "the CV splits fit in memory", but has some other nice features (e.g. the search of parameters exploits dask, re-using already-computed elements as much as possible). 

The cross-validation strategy employed in the grid-search can either come from `dask-ml` (but basically, only `KFold` is available) or from other sources: cross-validators from scikit-learn are supported, but one can also pass an iterator "yielding (train, test) splits as arrays of indices" [(see GridSearchCV docs, `cv` parameter)](https://ml.dask.org/modules/generated/dask_ml.model_selection.GridSearchCV.html#dask-ml-model-selection-gridsearchcv). Does this provide enough flexibility to pass in a user-defined CV strategy? For the specific of the Jackknife method, where you want to be able to specify groups of samples that are never split between train and test sets (e.g. all samples from a given year),   one could also use `sklearn.model_selection.GroupKFold`, where samples with the same group labels are never separated.  

Do you think that `GridSearchCV` should also be customisable?

Few other comments on the notebook, loosely related to the topic of the issue:

* the `pluck` method of dask bags allows one to unpack elements from a bag of iterables (e.g. to convert a bag of tuples in a tuple of bags). Thus one can do:
  ```python
  train_bag = train_test_bag.pluck(0)
  test_bag = train_test_bag.pluck(1)
  ```
* I think that for use cases that easily fit into memory, chunking in space and time probably only adds overhead to the training. One can still use the parallelisation over models and the parallelisation within the grid search for optimal parameters. I guess nothing really need to be changed, but I would make sure it is optional to convert the initial variables into dask arrays (and probably leave the default to keep numpy arrays). As a matter of fact, dask arrays are probably anyway converted back into numpy arrays when being passed to `sklearn.svm.SVR`.
* Especially for use cases that easily fit into memory, I would persist the training and test sets before moving on to the search for optimal parameters. Otherwise, you almost certainly repeat the first few steps (reading data, do the splitting) when having to use the test set at the end.

**Comment by rogerkuou** [comment]
Indeed for flexibility, let's do not create a function for this. Instead let's give some examples to do it.
Created #87 for following up.
